### ISSUE TITLE
don't always activate the BuildVision window

### ISSUE THREAD

**Issue by muellerto** [StefanKert/BuildVision#44]
The following is related to the activation of the BuildVision window when the build process starts. Sometimes, when I have a damned stubborn compilation error, I just compile a single C++ file several times (after modifications), without linking. In this case I want rather see the Output window. In this situation the BuildVision window does rather hide informations I want to see, again and again. Question: would it be possible to avoid the activation of the BuildVision window when I compile just a single file? Better: I suggest to activate the BuildVision window ONLY when the user builds the entire solution, not when he compiles a single file and also not when he builds a single project of the solution.

**Comment by LSMetag** [comment]
Yes, I think the BuildVision window should open only for Solution/project builds. Not for files.

**Comment by StefanKert** [comment]
So to get this right: C++ allows to compile single files in Visual Studio? I have not worked a lot with C++ and Visual Studio so I am going to need some help on this. If VS allows compiling single files we should definitely think about adding a setting that allows you to define what happens when compiling single files. 

For C# there are already enough settings to define what makes the window show up. 

Can anyone of you get me some screenshots how this looks? Or can you provide an example project, so I can try out what happens when I build a single file? I bet there are some flags that get passed to identify this and I will have to check this flags to let the window show / or not.

Thanks!

**Comment by muellerto** [comment]
Yes, Stefan, that's my use case: I want to compile just one single cpp file into an obj file, not an entire project and the the entire solution. The linker is not involved, just the compiler. This is a typical task, I do this every day. In C++ you have a context menu item called "Compile". You have also a button in the Build toolbar for this.

What the happens is neither a Project nor a Solution build. It's something different which should not interest BuildVision at all. BuildVision has no useful function in this case except showing the final result: successful or not. But what the user much more needs is the Build output. That's why he started this.

Again: if the user compiles a single file different from a Project or the Solution BuildVision should activate the Output window instead of the BuildVision Window.
